src/main/java/walle/FileParser.java

```diff
@@ -0,0 +1,92 @@
+package walle;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class FileParser {
+    public static int readFileContents(String filePath, ArrayList<Task> tasks) throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        int listSize = 0;
+        while (s.hasNext()) {
+            String[] taskString = s.nextLine().split(",");
+            boolean isDone = taskString[0].equals("1") ? true : false;
+            String taskType = taskString[1];
+            String taskDescription = taskString[2];
+
+            if (taskType.equals("T")) {
+                tasks.add(new Todo(taskDescription));
+            } else if (taskType.equals("D")) {
+                String dueDate = taskString[3];
+                tasks.add(new Deadline(taskDescription, dueDate));
+            } else if (taskType.equals("E")) {
+                String fromDate = taskString[3];
+                String toDate = taskString[4];
+                tasks.add(new Event(taskDescription, fromDate, toDate));
+            }
+
+            if (isDone) {
+                tasks.get(listSize).markAsDone();
+            }
```

Comment:
There are many magic literals here, consider using constants instead or if not possible, then add some concise comments on what is happening here.